javaslang/src/main/java/javaslang/collection/PriorityQueue.java

```diff
@@ -544,210 +544,214 @@ public boolean equals(Object o) {
+     * *      if rank t1 < rank t2 then      t1 :: meldUniq (ts1, t2 :: ts2)
+     * *      else if rank t2 < rank t1 then t2 :: meldUniq (t1 :: ts1, ts2)
+     * *      else                           ins (link (t1, t2), meldUniq (ts1, ts2))
+     **/
+    private static <T> Seq<Node<T>> meldUnique(Comparator<? super T> comparator, Seq<Node<T>> forest1, Seq<Node<T>> forest2) {
+        if (forest1.isEmpty()) {
+            return forest2;
+        } else if (forest2.isEmpty()) {
+            return forest1;
+        } else {
+            final Node<T> tree1 = forest1.head(), tree2 = forest2.head();
+
+            if (tree1.rank == tree2.rank) {
+                final Node<T> tree = tree1.link(comparator, tree2);
+                final Seq<Node<T>> forest = meldUnique(comparator, forest1.tail(), forest2.tail());
+                return ins(comparator, tree, forest);
+            } else {
+                if (tree1.rank < tree2.rank) {
+                    final Seq<Node<T>> forest = meldUnique(comparator, forest1.tail(), forest2);
+                    assert forest.isEmpty() || tree1.rank < forest.head().rank;
+                    return tree1.appendTo(forest);
                 } else {
-                    nonZeroRank = initialForestHead.appendTo(nonZeroRank);
+                    final Seq<Node<T>> forest = meldUnique(comparator, forest1, forest2.tail());
+                    assert forest.isEmpty() || tree2.rank < forest.head().rank;
+                    return tree2.appendTo(forest);
                 }
             }
-            return meld(comparator, nonZeroRank, zeroRank);
         }
+    }
 
-        /**
-         * fun insert (x, ts as t1 :: t2 :: rest) =
-         * *     if rank t1 = rank t2 then skewLink(Node(x,0,[]),t1,t2) :: rest
-         * *     else                      Node (x,0,[]) :: ts
-         * * | insert (x, ts) =            Node (x,0,[]) :: ts
-         **/
-        static <T> Seq<Node<T>> insert(SerializableComparator<? super T> comparator, T element, Seq<Node<T>> forest) {
-            final Node<T> tree = Node.of(element, 0, List.empty());
-            if (forest.size() >= 2) {
-                final Seq<Node<T>> tail = forest.tail();
-                final Node<T> t1 = forest.head(), t2 = tail.head();
-                if (t1.rank == t2.rank) {
-                    return tree.skewLink(comparator, t1, t2).appendTo(tail.tail());
-                }
-            }
-            return tree.appendTo(forest);
-        }
+    /* Based on http://www.brics.dk/RS/96/37/BRICS-RS-96-37.pdf */
+    static final class Node<T> implements Serializable {
+
+        private static final long serialVersionUID = 1L;
 
-        /** fun meld (ts, ts') = meldUniq (uniqify ts, uniqify ts') */
-        static <T> Seq<Node<T>> meld(SerializableComparator<? super T> comparator, Seq<Node<T>> source, Seq<Node<T>> target) {
-            return meldUnique(comparator, uniqify(comparator, source), uniqify(comparator, target));
+        final T root;
```

Comment:
\[Checkstyle\] ERROR: Variable 'root' must be private and have accessor methods\.